bfvmm/include/hve/arch/intel_x64/vcpu.h

```diff
@@ -347,61 +351,36 @@ class vcpu : public bfvmm::vcpu
     VIRTUAL void execute_wrcr4();
 
     //--------------------------------------------------------------------------
-    // CPUID
+    // vCPU CPUID Interface Implementation
     //--------------------------------------------------------------------------
 
-    /// Add CPUID Handler
-    ///
-    /// Add a delegate to handle a CPUID VM exit. Your handler should always
-    /// return false unless you need to override the default behavior of the
-    /// base hypervisor. For more information, please see the CPUID VM exit
-    /// handler for details.
-    ///
-    /// @expects
-    /// @ensures
-    ///
-    /// @param leaf the leaf to call d on
-    /// @param d the delegate to call on the vm exit
-    ///
-    VIRTUAL void add_cpuid_handler(
-        cpuid_handler::leaf_t leaf, const handler_delegate_t &d);
+    /// @cond
 
-    /// Add Emulate
-    ///
-    /// Emulate the exeuction of the CPUID instruction. For more information
-    /// please see the CPUID VM exit handler for details. Generally speaking,
-    /// you should typically use add_handler() instead.
-    ///
-    /// @expects
-    /// @ensures
-    ///
-    /// @param leaf the leaf to call d on
-    /// @param d the delegate to call on the vm exit
-    ///
-    VIRTUAL void add_cpuid_emulator(
-        cpuid_handler::leaf_t leaf, const handler_delegate_t &d);
+    void cpuid_add_handler(
+        vcpu_cpuid_interface::leaf_t leaf, const handler_delegate_t &d) OVERRIDE
+    { m_cpuid_handler.add_handler(leaf, d); }
 
-    /// Execute CPUID
-    ///
-    /// Executes the CPUID instruction, and populates the vCPU's registers.
-    /// For more information, please see the CPUID VM exit handler's
-    /// documentation.
-    ///
-    /// @expects
-    /// @ensures
-    ///
-    VIRTUAL void execute_cpuid();
+    void cpuid_add_emulator(
+        vcpu_cpuid_interface::leaf_t leaf, const handler_delegate_t &d) OVERRIDE
+    { m_cpuid_handler.add_emulator(leaf, d); }
 
-    /// Enable Whitelisting
-    ///
-    /// Ensures that if a VM exit occurs, that an emulator must be registered
-    /// for the exit. If an emulator is not registered, the VM exit is
-    /// reported as unhandled.
-    ///
-    /// @expects
-    /// @ensures
-    ///
-    VIRTUAL void enable_cpuid_whitelisting() noexcept;
+    void cpuid_execute() OVERRIDE
+    { m_cpuid_handler.execute(this); }
+
+    void cpuid_emulate(uint64_t rax, uint64_t rbx, uint64_t rcx, uint64_t rdx) OVERRIDE {
```

Comment:
Astyle is formatting this function really weird on my environment, it won't let me move the opening ```{``` to the next line

Reply:
see comments above and below. The issue is you are using a macro which will not be needed when this is fixed.